### PR TITLE
Deploy by tech

### PR DESCRIPTION
**Pull request by wawax**
Ability to deploy packages on demand, not only for a self service person.

Add informations about packages available and being installed in the task tab of a computer.
Also add the possiblity, when user is using the standard interface to select if wakeup should be local (calling the agent as 127.0.0.1) or remote, from the server.

### CODE REVIEW

File: inc/deploypackage.class.php [hunk 1]
```diff
@@ -1401,32 +1452,66 @@ function showPackageForMe($users_id) {
...
+      foreach ($mycomputers as $mycomputers_id => $data) {
+         $my_packages[$mycomputers_id] = [];
       }
+
+      //Get packages used for the user or a specific computer
       $packages_used = $this->getMyDepoyPackages($my_packages, $users_id);
 
-      $packages = $this->canUserDeploySelf();
+      //Get packages that a the user can deploy
+      $packages      = $this->canUserDeploySelf();
```
Comment (orthagh) [inline inc/deploypackage.class.php]: clear spaces before '=', i don't see any lines where this alignment is needed

File: inc/deploypackage.class.php [hunk 2]
```diff
@@ -1452,97 +1537,148 @@ function deployToComputer($computers_id, $packages_id, $users_id) {
...
-             'entities_id'             => $computer->fields['entities_id'],
-             'reprepare_if_successful' => 0
-         );
+
+         //Add the new task
+         $input = [
+                   'name'                    => '[deploy on demand] '.$this->fields['name'],
+                   'entities_id'             => $computer->fields['entities_id'],
+                   'reprepare_if_successful' => 0,
+                   'is_deploy_on_demand'      => 1
```
Comment (orthagh) [inline inc/deploypackage.class.php]: alignement

File: inc/deploypackage.class.php [hunk 3]
```diff
@@ -1452,97 +1537,148 @@ function deployToComputer($computers_id, $packages_id, $users_id) {
...
+                   'enduser'     => exportArrayToDB([$users_id  => [$computers_id]])
+                  ];
          $pfTaskJob->add($input);
-         $input = array(
-             'id'        => $tasks_id,
-             'is_active' => 1,
-         );
+
+         //Enable the task
+         $input = ['id' => $tasks_id, 'is_active' => 1];
```
Comment (orthagh) [inline inc/deploypackage.class.php]: Why not activating the task on the previous "->add($input)"

File: inc/menu.class.php [hunk 4]
```diff
@@ -387,18 +362,11 @@ static function displayMenu($type = "big") {
                                  "/plugins/fusioninventory/front/collectrule.php";
       }
 
-      if (Session::haveRight("plugin_fusioninventory_rulecollect", READ)) {
-         $a_menu[6]['name'] = __('Dictionnary of computer architectures', 'fusioninventory');
-         $a_menu[6]['pic']  = $CFG_GLPI['root_doc']."/plugins/fusioninventory/pics/menu_rules.png";
-         $a_menu[6]['link'] = $CFG_GLPI['root_doc'].
-                                 "/plugins/fusioninventory/front/ruledictionnarycomputerarch.php";
-      }
-
```
Comment (orthagh) [inline inc/menu.class.php]: You removed computer architectures rules from menu ?

File: inc/task.class.php [hunk 5]
```diff
@@ -768,10 +780,87 @@ static function cronTaskscheduler() {
...
+    * @return arrray of information
+   **/
+   static function cronInfo($name) {
+
+      switch ($name) {
+         case 'taskScheduler' :
+            return array('description' => __('FusionInventory task scheduler'));
+
+         case 'cleanOnDemand' :
+            return array('description' => __('Clean on deman deployment tasks'));
```
Comment (orthagh) [inline inc/task.class.php]: typo, deman -> demand

File: inc/menu.class.php (hunk 4 above)
Comment (wawax) [inline inc/menu.class.php]: yes, not needed anymore as it's been transfered into glpi core since 9.1

**Pull request merged**